crypto/bls_test.go

```diff
@@ -704,3 +705,54 @@ func BenchmarkVerifySignatureManyMessages(b *testing.B) {
 	}
 	b.StopTimer()
 }
+
+// Bench of all aggregation functions
+func BenchmarkAggregate(b *testing.B) {
+	// random message
+	input := make([]byte, 100)
+	_, _ = mrand.Read(input)
+	// hasher
+	kmac := NewBLSKMAC("bench tag")
+	sigsNum := 1000
+	sigs := make([]Signature, 0, sigsNum)
+	sks := make([]PrivateKey, 0, sigsNum)
+	pks := make([]PublicKey, 0, sigsNum)
+	seed := make([]byte, KeyGenSeedMinLenBLSBLS12381)
+
+	// create the signatures
+	for i := 0; i < sigsNum; i++ {
+		_, _ = mrand.Read(seed)
+		sk, _ := GeneratePrivateKey(BLSBLS12381, seed)
+		s, _ := sk.Sign(input, kmac)
+		sigs = append(sigs, s)
+		sks = append(sks, sk)
+		pks = append(pks, sk.PublicKey())
+	}
```

Comment:
Suggest to add a `require.NoError(t, ...)` sanity check for all methods returning an error.

Reply:
Those are used during a benchmark function `func BenchmarkAggregate(b *testing.B)` so I can't use `require`.
I've added error tests and panic instead.